core/client/fs/src/main/java/alluxio/client/file/MetadataCache.java

```diff
@@ -118,9 +127,19 @@ public void put(String path, URIStatus status) {
    * @param statuses the list status results
    */
   public void put(AlluxioURI dir, List<URIStatus> statuses) {
-    mCache.put(dir.getPath(), new CachedItem(statuses));
-    for (URIStatus status : statuses) {
-      mCache.put(status.getPath(), new CachedItem(status));
+    try {
+      CachedItem item = mCache.get(dir.getPath(), new Callable<CachedItem>() {
+        @Override
+        public CachedItem call() throws Exception {
+          return new CachedItem();
+        }
+      });
+      item.setDirStatuses(statuses);
+      for (URIStatus status : statuses) {
+        put(status.getPath(), status);
+      }
+    } catch (Exception e) {
```

Comment:
Generally, it's not a good practice to catch `Exception` - we should really only catch exceptions that we know can be thrown. Is there a list of exceptions you know could be thrown here?